### PR TITLE
[Calling][Feature] Custom Button Injection Button 

### PR DESCRIPTION
**Pull request by ShaunaSong**
## Purpose

* Custom control bar button 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

## How to Test

* Open...
* Click on...

## What to Check

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### CODE REVIEW

**Review by iaulakh — DISMISSED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallCompositeOptions.swift [hunk 1]
```diff
@@ -68,4 +69,48 @@ public struct CallCompositeOptions {
...
+    /// - Parameter callingScreenOrientation: callingScreenOrientation for specifying
+    ///  callingScreenOrientation customization. Default value is `nil`.
+    /// - Parameter enableMultitasking: enables user to navigate in the application
+    ///  while on the call. Default value is `false`.
+    /// - Parameter enableSystemPictureInPictureWhenMultitasking: enables syspem Picture-in-Picture while
+    ///  enableMultitasking is on and user navigates away from call view. Default value is `false`.
+    /// - Parameter callKitOptions: CallKitOptions for specifying CallKit customization. Default value is `nil`.
+    /// - Parameter displayName: Display name for the user. Default value is `nil`.
+    /// - Parameter disableInternalPushForIncomingCall: Disables internal push for incoming call.
+    /// - Parameter userId: User Identifier is used for local avatar injection. Default value is `nil`
```
Comment (iaulakh) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallCompositeOptions.swift:89]: Not matching args

Comment (iaulakh) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallCompositeOptions.swift:89]: capabilitiesChangedNotificationMode

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallScreenControlBarOptions.swift [hunk 2]
```diff
@@ -10,10 +10,42 @@ public struct CallScreenControlBarOptions {
...
+    public let liveCaptionsToggleButtonOptions: ButtonOptions?
+    public let spokenLanguageButtonOptions: ButtonOptions?
+    public let captionsLanguageButtonOptions: ButtonOptions?
+    public let shareDiagnosticsButtonOptions: ButtonOptions?
+    public let reportIssueButtonOptions: ButtonOptions?
+    public let customButtons: [CustomButtonOptions]
+
     /// Initializes an instance of CallScreenControlBarOptions.
     /// - Parameter leaveCallConfirmationMode: Whether to enable or disable the leave call confirmation.
     ///                                           Default is enabled.
```
Comment (iaulakh) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallScreenControlBarOptions.swift:26]: missing Parameter

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift [hunk 3]
```diff
@@ -40,22 +42,24 @@ public struct LocalOptions {
     ///   - skipSetupScreen: Indicates whether to bypass the setup screen.
     ///   - audioVideoMode: The desired audio/video mode for the call.
     ///   - captionsOptions: Configuration for captions
```
Comment (iaulakh) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift:44]: missing setupScreenOptions: SetupScreenOptions? = nil,
                callScreenOptions: CallScreenOptions? = nil)

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/SetupScreenOptions.swift [hunk 4]
```diff
@@ -9,12 +9,29 @@ import Foundation
...
     /// Creates an instance of SetupScreenOptions with related options.
     /// - Parameter cameraButtonEnabled: enables camera button on the setup screen.
     /// - Parameter microphoneButtonEnabled: enables microphone button on the setup screen.
+    @available(*, deprecated, message: "Use init with ButtonsOptions arguments.")
     public init(cameraButtonEnabled: Bool = true, microphoneButtonEnabled: Bool = true) {
         self.cameraButtonEnabled = cameraButtonEnabled
         self.microphoneButtonEnabled = microphoneButtonEnabled
     }
+    /// Creates an instance of SetupScreenOptions with related options.
+    /// - Parameter cameraButton: camera button options.c
```
Comment (iaulakh) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/SetupScreenOptions.swift:25]: spelling extra c

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/SetupScreenOptions.swift [hunk 5]
```diff
@@ -9,12 +9,29 @@ import Foundation
...
+    /// - Parameter microphoneButton: microphone button options.
+    /// - Parameter audioDeviceButton: audio device button options.
+    public init(cameraButton: ButtonOptions? = nil,
+                microphoneButton: ButtonOptions? = nil,
+                audioDeviceButton: ButtonOptions? = nil) {
+        self.cameraButtonEnabled = true
+        self.microphoneButtonEnabled = true
```
Comment (iaulakh) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/SetupScreenOptions.swift:32]: these two should be part of input args?

**Review by Mohtasim — APPROVED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift [hunk 6]
```diff
@@ -40,22 +42,26 @@ public struct LocalOptions {
...
-    ///   This role will determine functionality of the Setup screen. The actual role will be syncronized
-    ///   with ACS services when a Rooms call starts.
```
Comment (iaulakh) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift:44]: I think we need to add back

**Review by pavelprystinka — APPROVED** [review]

**Review by iaulakh — APPROVED** [review]

**Pull request merged**